srcmedia/ts/lib/input.test.ts

```diff
@@ -32,54 +56,75 @@ describe('RxTextInput', () => {
         document.body.innerHTML = `<input type="text">`
     })
 
-    it('updates when the user types into it', done => {
-        const element = document.querySelector('input[type=text]') as HTMLInputElement
-        const rti = new RxTextInput(element)
-        rti.state.subscribe(state => { // set up so that test completes once state is updated
-            expect(state).toEqual({ value: 'mysearch' })
-            expect(element.value).toBe('mysearch')
-            done() // done() is needed for async tests so we know when to expect the result
-        })
-        element.value = 'mysearch'
-        element.dispatchEvent(new Event('input')) // fake an input event as though the user typed
+    it('publishes its value as an observable, starting with initial value', () => {
+        const $input = document.querySelector('input') as HTMLInputElement
+        const rti = new RxTextInput($input)
+        const value = jest.fn()
+        rti.value$.subscribe(value)
+        expect(value).toHaveBeenCalledTimes(1) // called on subscription with current value
+        expect(value).toHaveBeenLastCalledWith('') // value is empty now
+        fakeValueChange($input, 'my text') // later we change the value
+        expect(value).toHaveBeenCalledTimes(2) // second call
+        expect(value).toHaveBeenLastCalledWith('my text') // updated with current value
     })
+})
+
+describe.skip('RxCheckboxInput', () => {
 
-    it('debounces rapid changes when the user is typing', done => {
-        const element = document.querySelector('input[type=text]') as HTMLInputElement
-        const rti = new RxTextInput(element)
-        const watcher = jest.fn()
-        rti.state.subscribe(watcher)
-        rti.state.subscribe(() => {
-            expect(watcher).toHaveBeenCalledTimes(1) // only called once
-            expect(watcher).toHaveBeenCalledWith({ value: 'mys' }) // only used the final value
-            done()
-        })
-        element.value = 'm'
-        element.dispatchEvent(new Event('input'))
-        element.value = 'my'
-        element.dispatchEvent(new Event('input'))
-        element.value = 'mys'
-        element.dispatchEvent(new Event('input'))
+    beforeEach(() => {
+        document.body.innerHTML = `<input type="checkbox" value="test!">`
     })
 
+    /*
+     * FIXME some problem here, potentially with jsdom...the 'change' event
```

Comment:
still an issue? doesn't look like the test is being skipped

Reply:
it should be getting skipped...the `.skip()` call is on the `describe()` block rather than the test itself though

Reply:
aha, I see that now. thanks.